File: Day_26/Q103.java
import java.util.*;

class Q103{
    static Scanner sc = new Scanner(System.in);

    //Initializing The Account
    static void input_bank_details(int arr[]){
        System.out.println("Enter Current Bank Balance :- ");
        arr[0] = sc.nextInt();

        System.out.println("Enter Pin :- ");
        arr[1] = sc.nextInt();
    }

    //Performing Withdrawal
    static void withdraw_amount(int arr[]){
        System.out.println("Enter Amount To Be Withdrawn :- ");

        int with_amt = sc.nextInt();

        if(with_amt <= 0){
            System.out.println("Invalid Amount");
            return;
        }

        if(with_amt > arr[0]){
            System.out.println("Withdrawal Amount Exceeds Balance!");
            return;
        }

        System.out.println("Enter The Pin :- ");

        int pass = sc.nextInt();

        if(pass != arr[1]){
            System.out.println("Invalid Pin");
            return;
        }

        arr[0] -= with_amt;

        System.out.println("Withdrawal Successful");
    }

    //Display Balance
    static void show_balance(int arr[]){
        System.out.println("Current Bank Balance :- " + arr[0]);
    }

    //ATM Simulation
    static void atm_simulation(int arr[]){
        int choice;

        do{
            System.out.println("\n===== ATM MENU =====");
            System.out.println("1. Check Bank Balance");
            System.out.println("2. Withdraw Amount");
            System.out.println("3. Exit");

            System.out.println("Enter Your Choice :- ");
            choice = sc.nextInt();

            switch(choice){
                case 1:
                    show_balance(arr);
                    break;

                case 2:
                    withdraw_amount(arr);
                    break;

                case 3:
                    System.out.println("Thank You For Using Our ATM Services!");
                    break;

                default:
                    System.out.println("Invalid Input");
            }

        }while(choice != 3);
    }

    public static void main(String[] args){
        int arr[] = new int[2];

        input_bank_details(arr);

        atm_simulation(arr);

        sc.close();
    }
}

File: Day_26/Q103.cpp
#include<bits/stdc++.h>
using namespace std;

//Initializing The Account
void input_bank_details(int arr[]){
    cout << "Enter Current Bank Balance :- ";
    cin >> arr[0];

    cout << "Enter Pin :- ";
    cin >> arr[1];
}

//Performing Withdrawal
void withdraw_amount(int arr[]){
    cout << "Enter Amount To Be Withdrawn :- ";

    int with_amt;
    cin >> with_amt;

    if(with_amt <= 0){
        cout << "Invalid Amount\n";
        return;
    }

    if(with_amt > arr[0]){
        cout << "Withdrawal Amount Exceeds Balance!\n";
        return;
    }

    cout << "Enter The Pin :- ";

    int pass;
    cin >> pass;

    if(pass != arr[1]){
        cout << "Invalid Pin\n";
        return;
    }

    arr[0] -= with_amt;

    cout << "Withdrawal Successful\n";
}

//Display Balance
void show_balance(int arr[]){
    cout << "Current Bank Balance :- " << arr[0] << "\n";
}

//ATM Simulation
void atm_simulation(int arr[]){
    int choice;

    do{
        cout << "\n===== ATM MENU =====\n";
        cout << "1. Check Bank Balance\n";
        cout << "2. Withdraw Amount\n";
        cout << "3. Exit\n";

        cout << "Enter Your Choice :- ";
        cin >> choice;

        switch(choice){
            case 1:
                show_balance(arr);
                break;

            case 2:
                withdraw_amount(arr);
                break;

            case 3:
                cout << "Thank You For Using Our ATM Services!\n";
                break;

            default:
                cout << "Invalid Input\n";
        }

    }while(choice != 3);
}

int main(){
    int arr[2];

    input_bank_details(arr);

    atm_simulation(arr);

    return 0;
}
